libs/CovidDatasets.py

```diff
@@ -277,26 +278,40 @@ def parse_country(country):
-                df = df.rename(columns=self._fieldname_map)
-                df = df.assign(**{self.DATE_FIELD: datetime.datetime.strptime(f.split('.')[0], '%m-%d-%Y')})
-                # Select out the subset of fields we care about
-                if self.COUNTY_FIELD not in df.columns:
-                    df[self.COUNTY_FIELD] = df[self.STATE_FIELD].dropna().apply(parse_county)
-                df = df[[self.DATE_FIELD, self.COUNTRY_FIELD, self.STATE_FIELD, self.COUNTY_FIELD, self.CASE_FIELD,
-                         self.DEATH_FIELD, self.RECOVERED_FIELD]]
-                # Parse the states from their longhand into their abbreviations
-                df[self.STATE_FIELD] = df[self.STATE_FIELD].dropna().apply(parse_state)
-                # Parse the 'US' entries into 'USA'
-                df[self.COUNTRY_FIELD] = df[self.COUNTRY_FIELD].apply(parse_country)
-                # Append the record dates by converting the file name into a datetime object
-                # Only process the csv files
-                day_reports.append(df)
+        snapshot_date = self._FIRST_JHU_DATE
+        while True:
+            csv_url = self._JHU_URL_TEMPLATE.format(snapshot_date.strftime('%m-%d-%Y'))
+            logging.info('Loading URL: {}'.format(csv_url))
+
+            # For each data file
+            try:
+                df = pd.read_csv(csv_url)
+            except urllib.error.HTTPError as e:
+                if e.code == 404:
+                    logging.info('Received a 404 for date {}. Ending iteration.'.format(snapshot_date))
+                    break
+                raise
+
+            df = df.rename(columns=self._fieldname_map)
+            snapshot_date_as_datetime = datetime.datetime.combine(snapshot_date, datetime.datetime.min.time())
+            df = df.assign(**{self.DATE_FIELD: snapshot_date_as_datetime})
+            # Select out the subset of fields we care about
+            if self.COUNTY_FIELD not in df.columns:
+                df[self.COUNTY_FIELD] = df[self.STATE_FIELD].dropna().apply(parse_county)
+            df = df[[self.DATE_FIELD, self.COUNTRY_FIELD, self.STATE_FIELD, self.COUNTY_FIELD, self.CASE_FIELD,
+                     self.DEATH_FIELD, self.RECOVERED_FIELD]]
+            # Parse the states from their longhand into their abbreviations
+            df[self.STATE_FIELD] = df[self.STATE_FIELD].dropna().apply(parse_state)
+            # Parse the 'US' entries into 'USA'
+            df[self.COUNTRY_FIELD] = df[self.COUNTRY_FIELD].apply(parse_country)
+            # Append the record dates by converting the file name into a datetime object
+            # Only process the csv files
+            day_reports.append(df)
+            snapshot_date += datetime.timedelta(days=1)
         full_report = pd.concat(day_reports)  # Concat said reports into a single DataFrame
         full_report = full_report.reset_index()
+
+        full_report.info()
```

Comment:
Just double-checking we want to leave this in?  (I'm not at all opposed; just flagging in case it was meant to be temporary for debugging)